File: python/h10.associative.py
#!/usr/bin/python
# initialize dictionary with key/value pairs
ages = {"bob": 34, "ed": 58, "steve": 32, "ralph": 23}

# append another set of key/value pairs into dictionary
ages.update({"deb": 46, "kate": 19})
 
# output the dictionary with key, one item per line
print "The ages are: "
for name in ages.keys():
    print " ages[%s]=%s" % (name, ages[name])

File: ruby/h10.associative.rb
#!/usr/bin/env ruby
# create empty hash
ages = Hash.new
# insert one element at a time
ages["bob"]   = 34
ages["ed"]    = 58
ages["steve"] = 32
ages["ralph"] = 23
ages["deb"]   = 46
ages["kate"]  = 19
 
# enumerate and print keys
puts "Keys (names):  #{ages.keys.join(" ")}"
# enumerate and print values
puts "Values (ages): #{ages.values.join(" ")}"